### PR TITLE
[Merged by Bors] - chore: use `+contextual`/`+decide` instead of `(config := { ... })`

### PR DESCRIPTION
**Pull request by kmill**
This is one pass at taking advantage of the new tactic configuration. There are plenty more uses of `(config := { ... })` in the library.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Qify.lean [hunk 1]
```diff
@@ -61,7 +61,7 @@ macro_rules
 | `(tactic| qify $[[$simpArgs,*]]? $[at $location]?) =>
   let args := simpArgs.map (·.getElems) |>.getD #[]
   `(tactic|
-    simp (config := {decide := false}) only [zify_simps, qify_simps, push_cast, $args,*]
+    simp -decide only [zify_simps, qify_simps, push_cast, $args,*]
```
Comment (kim-em) [inline Mathlib/Tactic/Qify.lean:64]: I guess we could just remove this now? But maybe this should happen elsewhen.

**Comment by kim-em** [comment]
bors d+

File: Mathlib/Tactic/Qify.lean (hunk 1 above)
Comment (kmill) [reply Mathlib/Tactic/Qify.lean:64]: Yeah, the `-decide` isn't necessary here anymore. I just didn't want to change the logic with this PR.

**Pull request closed without merging**